tests/nativeapp/test_project_model.py

```diff
@@ -173,3 +174,29 @@ def test_project_model_falls_back_to_current_role(
 
     assert project.app_role == CURRENT_ROLE
     assert project.package_role == CURRENT_ROLE
+
+
+@pytest.mark.parametrize("project_definition_files", ["minimal"], indirect=True)
+def test_bundle_context_from_project_model(project_definition_files: List[Path]):
+    project_defn = load_project(project_definition_files).project_definition
+    project_dir = Path().resolve()
+    project = NativeAppProjectModel(
+        project_definition=project_defn.native_app,
+        project_root=project_dir,
+    )
+
+    actual_bundle_ctx = project.get_bundle_context()
+
+    expected_bundle_ctx = BundleContext(
+        default_app_package("minimal"),
+        [
+            PathMapping(src="setup.sql", dest=None),
+            PathMapping(src="README.md", dest=None),
+        ],
+        project_dir,
+        Path(project_dir / "output" / "bundle"),
+        Path(project_dir / "output" / "deploy"),
+        Path(project_dir / "output" / "deploy" / "__generated"),
+    )
+
+    assert actual_bundle_ctx == expected_bundle_ctx
```

Comment:
as above, getting worried that the order of values here could become inconsistent. I'd prefer to have the text somehow verify that asking for the deploy root as the right value, etc.